### PR TITLE
Update the Snowflake, leafSnowflake, leafSegment property configuration requirements

### PR DESCRIPTION
**Pull request by haetao**

### CODE REVIEW

File: document/current/content/manual/sharding-jdbc/configuration/config-java.en.md [hunk 1]
```diff
@@ -283,7 +283,39 @@ The implementation class of `ShardingStrategyConfiguration`, used to configure n
...
-| props             | Properties                   | Property configuration, note:<br>To use the SNOWFLAKE algorithm, you need to configure the `worker.id` and `max.tolerate.time.difference.milliseconds` properties<br> To use the LEAF_SEGMENT algorithm, you need to configure: required `serverList`, `leafKey` and optional `initialValue`, `step`, `digest`, `registryCenterType`<br> To use the LEAF_SNOWFLAKE algorithm, you need to configure: required `serverList`, `serviceId` and optional `maxTimeDifference`, `digest`, `registryCenterType`|
+| props             | Properties                   | The Property configuration of  `SNOWFLAKE/LEAF_SEGMENT/LEAF_SNOWFLAKE` key generator |
+
+Property configuration that can include these properties of the key generators
+
+###### SNOWFLAKE
+  
+  | *Name*         | *DataType* |    *ConfigurationLevel*     | *Explanation* |
+  | ------------------- | -------- | ---------------- | ------------------ |
+  | worker.id(?)           | long |    optional | The unique id for working machine , the default is `0` if you do not configure it. |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.en.md]: `if you do not configure it` can be removed

File: document/current/content/manual/sharding-jdbc/configuration/config-java.en.md [hunk 2]
```diff
@@ -283,7 +283,39 @@ The implementation class of `ShardingStrategyConfiguration`, used to configure n
...
+  | max.tolerate.time.difference.milliseconds(?)        | long |  optional | The max tolerate time for clock rollback, its time unit is milliseconds, the default is `10` if you do not configure it. |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.en.md]: `clock rollback` should be `different server's time difference`?

**Comment by haetao** [comment]
I adjust the format of these tables  and modify some sentence , make them easier  to understand.

File: document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md [hunk 3]
```diff
@@ -288,15 +288,49 @@ ShardingStrategyConfiguration的实现类，用于配置不分片的策略。
 | *名称*             | *数据类型*                    | *说明*                                                                         |
 | ----------------- | ---------------------------- | ------------------------------------------------------------------------------ |
 | column            | String                       | 自增列名称                                                                      |
-| type              | String                       | 自增列值生成器类型，可自定义或选择内置类型：SNOWFLAKE/UUID/LEAF_SEGMENT                          |
-| props             | Properties                   | 属性配置, 注意：<br>使用SNOWFLAKE算法，需要配置worker.id与max.tolerate.time.difference.milliseconds属性<br>使用LEAF_SEGMENT算法，需要配置：必填项serverList，leafKey和选填项initialValue，step，digest，registryCenterType<br>使用LEAF_SNOWFLAKE算法，需要配置：必填项serverList，serviceId和选填项maxTimeDifference，digest，registryCenterType|  
+| type              | String                       | 自增列值生成器类型，可自定义或选择内置类型：SNOWFLAKE/UUID/LEAF_SEGMENT             |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md]: missing type `LEAF_SNOWFLAKE `

File: document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md [hunk 4]
```diff
@@ -288,15 +288,49 @@ ShardingStrategyConfiguration的实现类，用于配置不分片的策略。
...
+
+属性配置项，可以为以下自增列值生成器的属性
+
+##### SNOWFLAKE
+  
+  | *名称*                                              | *数据类型* |    *配置级别*     | *说明*                                                                                                   |
+  | --------------------------------------------------- | ----------| ---------------- | ------------------------------------------------------------------------------------------------------- |
+  | worker.id(?)                                        | long      |    选填          | 工作机器唯一id ，默认为0                                                                                   |
+  | max.tolerate.time.difference.milliseconds(?)        | long      |    选填          | 最大容忍时钟回退时间，单位：毫秒。默认为10毫秒                                                               |
+  | max.vibration.offset(?)                             | int       |    选填          | 最大抖动上限值，范围[0 , 4096)，默认为1。注：若使用此算法生成值作分片值，建议配置此属性。因为默认配置下，此算法在不同毫秒内所生成的key取模2^n (2^n一般为您的分库或分表数) 之后结果总为为0或1。为防止上述分片问题，建议您将此属性值配置为(2^n)-1 |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md]: `[0 , 4096)` should be `[0, 4096)`

File: document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md [hunk 5]
```diff
@@ -288,15 +288,49 @@ ShardingStrategyConfiguration的实现类，用于配置不分片的策略。
...
+  | ------------------------------------ | -------------- | ----------------- | ------------------------------------------------------------------------------------ |
+  | server.list(?)                       | String          |    必填           | 连接注册中心服务器的列表，包括IP地址和端口号，多个地址用逗号分隔 如: host1:2181,host2:2181 |
+  | leaf.key(?)                          | String          |    必填           | 最大号段id所对应的key                                                                 |
+  | leaf.segment.id.initial.value(?)     | long            |    选填           | 号段id初始值，默认为1                                                                 |
+  | leaf.segment.step(?)                 | long            |    选填           | 每次分配的号段步长，默认为10000                                                        |
+  | registry.center.digest(?)            | String          |    选填           | 连接注册中心的权限令牌。缺省为不需要权限验证                                             |
+  | registry.center.type(?)              | String          |    选填           | 注册中心类型，默认为"zookeeper"                                                       |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.cn.md]: "zookeeper"  should be `zookeeper`

File: document/current/content/manual/sharding-jdbc/configuration/config-java.en.md [hunk 6]
```diff
@@ -283,7 +283,39 @@ The implementation class of `ShardingStrategyConfiguration`, used to configure n
 | ----------------- | ---------------------------- | ------------------------------------------------------------------------------------------- |
 | column            | String                       | Column name of key generator                                                                |
 | type              | String                       | Type of key generator，use user-defined ones or built-in ones, e.g. SNOWFLAKE, UUID         |
-| props             | Properties                   | Property configuration, note:<br>To use the SNOWFLAKE algorithm, you need to configure the `worker.id` and `max.tolerate.time.difference.milliseconds` properties<br> To use the LEAF_SEGMENT algorithm, you need to configure: required `serverList`, `leafKey` and optional `initialValue`, `step`, `digest`, `registryCenterType`<br> To use the LEAF_SNOWFLAKE algorithm, you need to configure: required `serverList`, `serviceId` and optional `maxTimeDifference`, `digest`, `registryCenterType`|
+| props             | Properties                   | The Property configuration of  `SNOWFLAKE/LEAF_SEGMENT/LEAF_SNOWFLAKE` key generator        |
```
Comment (terrymanu) [inline document/current/content/manual/sharding-jdbc/configuration/config-java.en.md]: The Property configuration of  `SNOWFLAKE/LEAF_SEGMENT/LEAF_SNOWFLAKE` key generator 

should be:

The Property configuration of  key generator

**Pull request merged**